src/components/header/_macro.spec.js

```diff
@@ -6,523 +6,348 @@ import axe from '../../tests/helpers/axe';
+        expect($('.ons-header__title-logo--large img').attr('src')).toBe('another-logo.svg');
+      });
+      test('THEN: renders provided small title logo on mobile', () => {
+        expect($('.ons-header__title-logo--small img').attr('src')).toBe('another-logo-small.svg');
+      });
     });
-
-    it('has the provided title logo custom classes', () => {
+    describe('WHEN: titleLogo: classes parameter is provided', () => {
       const $ = cheerio.load(
         renderComponent('header', {
           ...EXAMPLE_HEADER_BASIC,
           titleLogo: { large: 'another-logo.svg', classes: 'custom-class another-custom-class' },
         }),
       );
-
-      expect($('.ons-header__title-logo--large').hasClass('custom-class')).toBe(true);
-      expect($('.ons-header__title-logo--large').hasClass('another-custom-class')).toBe(true);
+      test('THEN: renders title logo with provided additional classes', () => {
+        expect($('.ons-header__title-logo--large').hasClass('custom-class')).toBe(true);
+        expect($('.ons-header__title-logo--large').hasClass('another-custom-class')).toBe(true);
+      });
     });
-
-    it('has the provided small title logo', () => {
+    describe('WHEN: description parameter is provided', () => {
       const $ = cheerio.load(
         renderComponent('header', {
           ...EXAMPLE_HEADER_BASIC,
-          titleLogo: { large: '<img src="another-logo.svg">', small: '<img src="another-logo-small.svg">' },
+          description: 'Header description',
         }),
       );
-
-      expect($('.ons-header__title-logo--small img').attr('src')).toBe('another-logo-small.svg');
-    });
-
-    it('displays the `description` text', () => {
-      const $ = cheerio.load(renderComponent('header', { ...EXAMPLE_HEADER_BASIC, description: 'Header description' }));
-
-      expect($('.ons-header__description').text()).toBe('Header description');
+      test('THEN: renders title with provided description', () => {
+        expect($('.ons-header__description').text()).toBe('Header description');
+      });
     });
-
-    it('renders a button with expected parameters', () => {
-      const faker = templateFaker();
-      const buttonSpy = faker.spy('button', { suppressOutput: true });
-
-      faker.renderComponent('header', {
-        ...EXAMPLE_HEADER_BASIC,
-        button: {
-          text: 'Save and sign out',
-          name: 'button-name',
-          attributes: {
-            a: 'b',
-          },
-          url: '#0',
-        },
+  });
+  describe('GIVEN: Params: button: SignOutButton', () => {
+    describe('WHEN: no button parameters are provided', () => {
+      const $ = cheerio.load(
+        renderComponent('header', {
+          ...EXAMPLE_HEADER_BASIC,
+        }),
+      );
+      test('THEN: renders title grid with gutterless class', () => {
+        const titleGridDiv = $('.ons-header__main .ons-container .ons-grid');
+        expect($(titleGridDiv).hasClass('ons-grid--gutterless')).toBe(true);
```

Comment:
Think the wording on this test needs to be a bit more clear that there is no button expected to be rendered and maybe we need to add a test to show that

Reply:
we are technically not testing for this though - that would be a separate test I think! We are just checking that if we don't provide any parameters for a button that the rest of title grid displays properly

Reply:
Yeah true, fair enough